src/test/kotlin/org/rust/ide/annotator/RsErrorAnnotatorTest.kt

```diff
@@ -1747,6 +1747,37 @@ class RsErrorAnnotatorTest : RsAnnotatorTestBase(RsErrorAnnotator::class.java) {
         }
     """)
 
+    fun `test duplicate enum discriminant const expr E0081`() = checkErrors("""
+        enum Bad {
+            <error descr="Discriminant value `3` already exists [E0081]">X = 1 + 5 - 3</error>,
+            <error descr="Discriminant value `3` already exists [E0081]">Y = 1 + 2</error>
+        }
+    """)
+
+    fun `test duplicate enum discriminant non-const expr E0081`() = checkErrors("""
+        fn foo() -> isize { 0 }
+        enum Good {
+            X = foo(),
+            Y = 0
+        }
+    """)
+
+    fun `test duplicate enum discriminant repr type valid range E0081`() = checkErrors("""
+        #[repr(i8)]
+        enum Bad {
+            <error descr="Discriminant value `-1` already exists [E0081]">X = -1</error>,
+            <error descr="Discriminant value `-1` already exists [E0081]">Y = -1</error>
+        }
+    """)
+
+    fun `test duplicate enum discriminant repr type invalid range E0081`() = checkErrors("""
+        #[repr(u8)]
+        enum Good {
```

Comment:
Why is this enum called `Good` when it is in fact invalid ?

Reply:
Maybe the name can be better.
But anyway, we don't have API to check overflows like here currently so we prefer don't show error annotations when we fail to evaluate an expression to avoid false-positive annotations.
We are going to improve our constant evaluator to handle such cases too